apps/dcos_overlay/src/dcos_overlay_poller.erl

```diff
@@ -205,69 +278,138 @@ match(_Attr, _List) ->
+    case dcos_overlay_netlink:ipaddr_replace(Pid, inet, ParsedVTEPIP, PrefixLen, VTEPNameStr) of
+        {ok, _} ->
+            lager:notice("Overlay VTEP address was configured, ~s => ~p", [VTEPName, VTEPIP]),
+            maybe_create_vtep_addr6(Pid, VTEPName, VTEPNameStr, VTEPIP6);
+        {error, Error}  ->
+            lager:error("Failed to replace address ~p for VTEP ~s due to ~p",
+                        [VTEPIP, VTEPNameStr, Error]),
+            {error, Error}
+    end.
+
+maybe_create_vtep_addr6(Pid, VTEPName, VTEPNameStr, VTEPIP6) ->
     case {application:get_env(dcos_overlay, enable_ipv6, true), VTEPIP6} of
-      {_, undefined} ->
-          ok;
-      {false, _} ->
-          lager:notice("Overlay network is disabled [ipv6], ~s => ~p", [VTEPName, VTEPIP6]);
-      _ ->
-          ok = try_enable_ipv6(VTEPName),
-          {ParsedVTEPIP6, PrefixLen6} = parse_subnet(VTEPIP6),
-          {ok, _} = dcos_overlay_netlink:ipaddr_replace(Pid, inet6, ParsedVTEPIP6, PrefixLen6, VTEPNameStr),
-          lager:notice("Overlay VTEP address was configured [ipv6], ~s => ~p", [VTEPName, VTEPIP6])
+        {_, undefined} ->
+            ok;
+        {false, _} ->
+            lager:notice("Overlay network is disabled [ipv6], ~s => ~p",
+                         [VTEPName, VTEPIP6]),
+            ok;
+        _ ->
+            ensure_vtep_addr6_created(Pid, VTEPName, VTEPNameStr, VTEPIP6)
     end.
 
+ensure_vtep_addr6_created(Pid, VTEPName, VTEPNameStr, VTEPIP6) ->
+    case try_enable_ipv6(VTEPName) of
+        ok ->
+            {ParsedVTEPIP6, PrefixLen6} = parse_subnet(VTEPIP6),
+            case dcos_overlay_netlink:ipaddr_replace(
+                   Pid, inet6, ParsedVTEPIP6, PrefixLen6, VTEPNameStr) of
+                {ok, _} ->
+                    lager:notice("Overlay VTEP address was configured [ipv6], ~s => ~p",
+                                 [VTEPNameStr, VTEPIP6]),
+                    ok;
+                {error, Error}  ->
+                    lager:error("Failed to replace address ~p for VTEP ~s due to ~p",
+                                [VTEPIP6, VTEPNameStr, Error]),
+                    {error, Error}
+            end;
+        {error, Error} ->
+            {error, Error}
+    end.
 
 try_enable_ipv6(IfName) ->
     Opt = <<"net.ipv6.conf.", IfName/binary, ".disable_ipv6=0">>,
     case dcos_net_utils:system([<<"/sbin/sysctl">>, <<"-w">>, Opt]) of
         {ok, _} ->
             ok;
         {error, Error} ->
-            lager:error(
-                "Couldn't enable IPv6 on ~s interface due to ~p",
-                [IfName, Error])
+            lager:error("Couldn't enable IPv6 on ~s interface due to ~p",
+                        [IfName, Error]),
+            {error, Error}
     end.
 
-maybe_add_ip_rule(#{<<"info">> := #{<<"subnet">> := Subnet},
-                    <<"backend">> := #{<<"vxlan">> := #{<<"vtep_name">> := VTEPName}}},
-                  #state{netlink = Pid}) ->
+maybe_add_ip_rule(Pid,
```

Comment:
```suggestion
maybe_add_ip_rule(Pid, Overlay) ->
    #{…} = Overlay,
```

Reply:
This function clause relies on pattern matching the way it is done right now. Otherwise, the other clause should be removed, and `case` be introduced into the first clause.

Reply:
Got it.